static/app/components/notificationActions/notificationActionManager.tsx

```diff
@@ -0,0 +1,219 @@
+            index={index}
+            defaultEdit={!action.id}
+            action={action}
+            recipientRoles={recipientRoles}
+            availableActions={availableServices[serviceType]}
+            pagerdutyIntegrations={pagerdutyIntegrations}
+            project={project}
+            onDelete={removeNotificationAction}
+            onUpdate={updateNotificationAction}
+          />
+        ))
+      );
+    });
+    return serviceActions;
+  }
+
+  function addNotificationAction(actionInstance: AvailableNotificationAction['action']) {
+    const updatedActions = cloneDeep(notificationActions);
+    updatedActions.push(actionInstance);
+    setNotificationActions(updatedActions);
+    updateAlertCount(parseInt(project.id, 10), updatedActions.length);
+  }
+
+  function removeNotificationAction(index: number) {
+    // Removes notif action from state using the index
+    const updatedActions = cloneDeep(notificationActions);
+    updatedActions.splice(index, 1);
+    setNotificationActions(updatedActions);
+    updateAlertCount(parseInt(project.id, 10), updatedActions.length);
+  }
+
+  function updateNotificationAction(index: number, updatedAction: NotificationAction) {
+    // Updates notif action from state using the index
+    const updatedActions = cloneDeep(notificationActions);
+    updatedActions.splice(index, 1, updatedAction);
+    setNotificationActions(updatedActions);
+  }
+
+  // The dropdown items for "Add Alert"
+  const getMenuItems = (): MenuItemProps[] => {
+    const menuItems: MenuItemProps[] = [];
+    Object.entries(availableServices).forEach(entry => {
+      const [serviceType, validActions] = entry;
+      if (validActions.length > 0) {
+        // Cannot have more than one Sentry notification
+        if (
+          serviceType === NotificationActionService.SENTRY_NOTIFICATION &&
+          actionsMap[serviceType].length === 1
+        ) {
+          return;
+        }
+
+        menuItems.push({
+          key: serviceType,
+          label: t(
+            'Send a %s notification',
+            serviceType !== NotificationActionService.SENTRY_NOTIFICATION
+              ? capitalize(serviceType)
+              : 'Sentry'
+          ),
+          onAction: () => addNotificationAction(validActions[0].action),
+        });
+      }
+    });
+    return menuItems;
+  };
+
+  function renderAddAlertButton() {
+    return (
+      <DropdownMenu
+        items={getMenuItems()}
+        trigger={triggerProps => (
+          <Button
+            {...triggerProps}
+            aria-label={t('Add Alert')}
+            size="xs"
+            icon={<IconAdd isCircled color="gray300" />}
+            onClick={e => {
+              e.stopPropagation();
+              e.preventDefault();
+
+              triggerProps.onClick?.(e);
+            }}
+          >
+            {t('Add Action')}
+            <StyledIconChevron direction="down" />
+          </Button>
+        )}
+      />
+    );
+  }
```

Comment:
```suggestion
  const addAlertButton = (
    <DropdownMenu
      items={getMenuItems()}
      trigger={triggerProps => (
        <Button
          {...triggerProps}
          aria-label={t('Add Alert')}
          size="xs"
          icon={<IconAdd isCircled color="gray300" />}
          onClick={e => {
            e.stopPropagation();
            e.preventDefault();

            triggerProps.onClick?.(e);
          }}
        >
          {t('Add Action')}
          <StyledIconChevron direction="down" />
        </Button>
      )}
    />
  );
```

no need for the function here, it's called every time anyway, you can just stick it in a variable.